tracer/src/Datadog.Trace/OTelMetrics/MetricState.cs

```diff
@@ -91,39 +97,51 @@ public void RecordMeasurementDouble(double value, ReadOnlySpan<KeyValuePair<stri
         }
     }
 
-    public void RecordMeasurementGaugeLong(long value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateGauge(value);
-
-    public void RecordMeasurementGaugeDouble(double value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateGauge(value);
-
-    public void RecordMeasurementHistogramLong(long value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateHistogram(value);
-
-    public void RecordMeasurementHistogramDouble(double value, ReadOnlySpan<KeyValuePair<string, object?>> tags)
-        => GetOrCreatePoint(tags).UpdateHistogram(value);
-
-    public IEnumerable<MetricPoint> GetMetricPoints() => _points.Values;
-
-    public MetricStreamIdentity GetIdentity() => _identity;
+    /// <summary>
+    /// Builds metric point snapshots and adds them to the provided list.
+    /// For Delta temporality, resets the running values after snapshotting.
+    /// </summary>
+    public void TryBuildPoints(List<MetricPoint> into)
```

Comment:
nit: You should remove the `Try` prefix. There's a standard C# convention to do `bool TrySomeOperation(out result)` and this goes against that convention

```suggestion
    public void BuildPoints(List<MetricPoint> into)
```